gh-85583: Add overview of formatted string literals (f-strings) to ``str`` (#132689)
python/cpython: Doc/library/stdtypes.rst

and are converted into regular :class:`str` objects.
For example:

.. doctest::

   >>> who = 'nobody'
   >>> nationality = 'Spanish'
   >>> f'{who.title()} expects the {nationality} Inquisition!'
   'Nobody expects the Spanish Inquisition!'

It is also possible to use a multi line f-string:

.. doctest::

   >>> f'''This is a string
   ... on two lines'''
   'This is a string\non two lines'

A single opening curly bracket, ``'{'``, marks a *replacement field* that
can contain any Python expression:

.. doctest::

   >>> nationality = 'Spanish'
   >>> f'The {nationality} Inquisition!'
   'The Spanish Inquisition!'

To include a literal ``{`` or ``}``, use a double bracket:

.. doctest::

   >>> x = 42
   >>> f'{{x}} is {x}'
   '{x} is 42'

Functions can also be used, and :ref:`format specifier <formatstrings>`:

.. doctest::

   >>> from math import sqrt
   >>> f'√2 \N{ALMOST EQUAL TO} {sqrt(2):.5f}'
   '√2 ≈ 1.41421'

Any non-string expression is converted using :func:`str`, by default:

.. doctest::

   >>> from fractions import Fraction
   >>> f'{Fraction(1, 3)}'
   '1/3'

To use an explicit conversion, use the ``!`` (exclamation mark) operator,

> followed by any of the valid formats, which are:

Reviewer: ```suggestion
followed by one of the following valid formats:
```
Minor suggestion, better flow IMO.
